packages/store/addon/-private/system/ds-model-store.ts

```diff
@@ -32,6 +30,18 @@ type NotificationManager = import('./record-notification-manager').default;
 type DSModel = import('../ts-interfaces/ds-model').DSModel;
 type ShimModelClass = import('./model/shim-model-class').default;
 type DSModelClass = import('@ember-data/model').default;
+type RecordDataClass = typeof import('@ember-data/record-data/-private').RecordData;
+
+// we can't greedily use require as this causes
+// an order of operations issue which gives us undefined
+let _RecordData: RecordDataClass;
```

Comment:
FWIW, I think the types are not quite correct here, I would expect:

```suggestion
let _RecordData: undefined | RecordDataClass;
```